src/test/kotlin/fi/metatavu/timebank/api/test/functional/resources/TestWiremockResource.kt

```diff
@@ -98,65 +110,81 @@ class TestWiremockResource: QuarkusTestResourceLifecycleManager {
-                .willSetStateTo("generatedStateTwo")
-                .willReturn(jsonResponse(objectMapper.writeValueAsString(TestData.getGeneratedForecastTimeEntryResponse(pageNumber = 1)), 200))
-        )
-        wireMockServer.stubFor(
-            get(urlPathEqualTo("/v4/time_registrations"))
-                .withQueryParam("pageNumber", matching("2"))
-                .inScenario("timesScenario")
-                .whenScenarioStateIs("generatedStateTwo")
-                .willSetStateTo(STARTED)
-                .willReturn(jsonResponse(objectMapper.writeValueAsString(TestData.getGeneratedForecastTimeEntryResponse(pageNumber = 2)), 200))
+            get(urlPathEqualTo("/v4/time_registrations/updated_after/${getPathParamDate(getThirtyDaysAgo())}"))
+                .inScenario(TIMES_SCENARIO)
+                .whenScenarioStateIs(STARTED)
+                .willReturn(jsonResponse(objectMapper.writeValueAsString(TestData.getForecastTimeEntryResponse(
+                    before = null,
+                    after = getThirtyDaysAgo()
+                )), 200))
         )
         wireMockServer.stubFor(
-            get(urlPathEqualTo("/v4/time_registrations/updated_after/20220501T000000"))
-                .inScenario("timesScenario")
+            get(urlPathEqualTo("/v4/time_registrations/updated_after/${getPathParamDate(getSixtyDaysAgo())}"))
+                .inScenario(TIMES_SCENARIO)
                 .whenScenarioStateIs(STARTED)
                 .willReturn(jsonResponse(objectMapper.writeValueAsString(TestData.getForecastTimeEntryResponse(
                     before = null,
-                    after = "2022-05-01"
+                    after = getSixtyDaysAgo()
                 )), 200))
         )
+    }
+
+    /**
+     * /v4/tasks -stubs
+     */
+    private fun tasksStubs(wireMockServer: WireMockServer) {
+        wireMockServer.stubFor(
+            get(urlPathEqualTo("/v4/tasks"))
+                .inScenario(TASKS_SCENARIO)
+                .whenScenarioStateIs(STARTED)
+                .willReturn(jsonResponse(objectMapper.writeValueAsString(TestData.getForecastTaskResponse()), 200))
+        )
         wireMockServer.stubFor(
-            get(urlPathEqualTo("/v4/time_registrations/updated_after/20220501T000000"))
-                .inScenario("timesScenario")
-                .whenScenarioStateIs("errorState")
+            get(urlPathEqualTo("/v4/tasks"))
+                .inScenario(TASKS_SCENARIO)
+                .whenScenarioStateIs(ERROR_STATE)
                 .willSetStateTo(STARTED)
                 .willReturn(jsonResponse(objectMapper.writeValueAsString(ForecastErrorResponse(
                     status = 401,
-                    message = "Failed to authenticate the request."
+                    message = "Failed to authenticate the request"
                 )), 401))
         )
     }
 
+    private fun getPathParamDate(date: LocalDate): String {
+        return "${date.toString().replace("-", "")}T000000"
```

Comment:
Format seems a bit odd, what is the story behind it? Also why are you not using DateTimeFormatter?

Reply:
Because I had no idea that something like DateTimeFormatter even exists... :+1: